eng/doc/MigrationGuide.md

```diff
@@ -0,0 +1,231 @@
+# Migration Guide: Microsoft build of Go
+
+This guide provides a high-level overview of migrating from Google's standard Go distribution to Microsoft's build of Go. The Microsoft build of Go is designed to be a drop-in replacement that adds FIPS 140-2 compliance capabilities while maintaining full compatibility with standard Go applications.
+
+## Quick Start
+
+**For most users:** The Microsoft build of Go works exactly like standard Go - no code changes required. Simply replace your Go installation and continue developing as usual. As of Go 1.25+, `systemcrypto` is enabled by default on Windows and Linux.
+
+**For FIPS compliance:** System crypto is already enabled by default. Configure runtime FIPS mode as needed for your specific requirements.
+
+**To opt out of system crypto:** Set `GOEXPERIMENT=nosystemcrypto` when building (not recommended for Microsoft internal use).
+
+## What's Different?
+
+The Microsoft build of Go includes patches that enable:
+- **FIPS 140-2 compliance** through platform-native crypto backends
+- **OpenSSL integration** on Linux
+- **CNG (Cryptography Next Generation)** on Windows  
+- **CommonCrypto/CryptoKit** on macOS
+- **No code changes required** - same Go APIs
+- **Runtime configuration** for FIPS mode
+
+## Migration Paths
+
+Choose your path based on your knowledge level and requirements:
```

Comment:
It's not clear to me if there are really different paths, or just some steps that aren't necessary if you aren't targeting FIPS/FedRAMP (very few, mainly review and testing!).